### ISSUE TITLE
PCC Drop in reduce_scatter / multiply / softmax with Specific Input Tensors (fp32)

### ISSUE THREAD

**Issue by hkwonTT** [tenstorrent/tt-metal#21722]
### Describe the bug

Significant PCC drop observed in the following operations: reduce_scatter, multiply, and softmax.
All tests were conducted using float32 (fp32) dtype.

```
reduce_scatter PCC: 0.6153663253974962  
multiply PCC: 0.7155594611774694  
softmax PCC: 0.0
```

The PCC values were computed by comparing the device output against PyTorch CPU output.

This issue only occurs with specific input tensors. When randomly generated tensors are used, the PCC values are acceptable.

The problematic input tensors come from the later stages of neural network models:
	•	reduce_scatter: from a multi-device LLaMA attention block
	•	multiply: from a single-device LLaMA attention block
	•	softmax: from a single-device MNIST model

FYI:
	•	No fp32 values became inf or NaN.
	•	However, many values became inf or NaN when the same inputs were cast to fp16.

### Steps to reproduce the issue

[testcase.zip](https://github.com/user-attachments/files/20071331/testcase.zip)

1. unzip attached file
2. `python testing_pcc_drop.py`
3. observe PCC

### Expected behavior

PCC should be almost 1

**Comment by hkwonTT** [comment]
@rtawfik01 
I’m not sure who the right person is to look into this issue. Please feel free to reassign it if appropriate.

**Comment by hkwonTT** [comment]
@rtawfik01 
I mentioned that eltwise add was also facing the same issue, but it turns out that it wasn’t. Never mind — thanks!

**Comment by rtawfik01** [comment]
@ncvetkovicTT @nvelickovicTT from the list above, we are aware of the eltwise multiply not doing float32 high fidelity because the dest accum flag was never set to true. For the other ops we need to investigate the accuracy drops.

**Comment by fahmedchanTT** [comment]
@cmaryanTT You mentioned this case needs to be broken down further, can you help with that?

**Comment by cmaryanTT** [comment]
@hkwonTT @fahmedchanTT - I've created sub issues. In the future, please note that these are three different unrelated OPs, and you should file independent issues for them

**Comment by bbradelTT** [comment]
@hkwonTT  The initial input was very sensitive to numerical stability. The PCC improves when passing in the flag. Please use
```
output_ttnn = ttnn.softmax(input_ttnn, -1, numeric_stable=True)
```
then you should get:
```
softmax PCC: 0.9999999999999997
```

This will be set to True by default hopefully soon.

You can get more accuracy in softmax in general also by adding fp32_dest_acc_en. Something like
```
        compute_kernel_config = ttnn.WormholeComputeKernelConfig(
        math_fidelity=ttnn.MathFidelity.HiFi4,
        math_approx_mode=False,
        fp32_dest_acc_en=True,
        packer_l1_acc=True,
        )
        output_ttnn = ttnn.softmax(input_ttnn, -1, compute_kernel_config=compute_kernel_config, numeric_stable=True)
```

**Comment by hkwonTT** [comment]
Thank you for all your support!
